### ISSUE TITLE
await-job-completion doesn't wait for full task lifecycle to be completed

### ISSUE THREAD

**Issue by lbradstreet** [onyx-platform/onyx#292]
Something I noticed in a new onyx-kafka test I completed:

``` clojure
(def results (doall (map (fn [_] (<!! out-chan)) (range 3))))

(onyx.api/kill-job peer-config job-id)
(onyx.api/await-job-completion peer-config job-id)

(kp/send-message producer (kp/message topic (.getBytes (pr-str {:n 4}))))
(kp/send-message producer (kp/message topic (.getBytes (pr-str {:n 5}))))
(kp/send-message producer (kp/message topic (.getBytes (pr-str {:n 6}))))
(kp/send-message producer (kp/message topic (.getBytes (pr-str :done))))
```

Messages {:n 4}, 5, 6 may still be handled by the job. This may or may not be a problem that we should solve. I can't think of a smart way to currently check for this, and perhaps we need another signal sent by the peers after the task is stopped.

**Comment by MichaelDrogalis** [comment]
I think this is an appropriate change. It may take a little wiggling around from where we send the signal, but this is worth it to me. Letting a piece of a lifecycle run outside the context of a job seems like a bug IMO.

**Comment by lbradstreet** [comment]
Cool, I agree.

**Comment by MichaelDrogalis** [comment]
Looking at this one. Would it be enough to move [this](https://github.com/onyx-platform/onyx/blob/15795a05102a33c0aa8f7a5bb55554080403c4bb/src/onyx/peer/task_lifecycle.clj#L744-L745) and [this](https://github.com/onyx-platform/onyx/blob/15795a05102a33c0aa8f7a5bb55554080403c4bb/src/onyx/peer/task_lifecycle.clj#L764) to [here](https://github.com/onyx-platform/onyx/blob/15795a05102a33c0aa8f7a5bb55554080403c4bb/src/onyx/peer/task_lifecycle.clj#L58)? I *believe* this is insufficient because simply appending the log entry doesn't mean the job is actually complete. Trying to find a minimal change to fix this one.

**Comment by neverfox** [comment]
Adding my confirmation. As @MichaelDrogalis pointed out to me, this is happening more often with changes to 0.9.7 that shut things down faster. It appeared to not be a problem in 0.9.6 but that was apparently luck of timing.

**Comment by lbradstreet** [comment]
Should be fixed by 0.10.
